realm/realm-library/src/main/java/io/realm/rx/RealmObservableFactory.java

```diff
@@ -222,4 +222,9 @@ public boolean equals(Object o) {
         return o instanceof RealmObservableFactory;
     }
 
+    @Override
+    public int hashCode() {
+        return 37;
```

Comment:
The class has no state that can be set from the outside, so any instance of this class is equal to any other instance of this class. A hardcoded hashCode is fine in that case :)